agent/agent.go

```diff
@@ -940,6 +945,10 @@ func (a *Agent) StopInputPlugin(input string, shouldUpdatePlugin bool) error {
 	close(statusChannel.stop)
 	close(statusChannel.stopped)
 
+	if shouldUpdateConfig {
+		a.Config.UpdateConfig(map[string]interface{}{}, input, "inputs", "STOP_PLUGIN")
```

Comment:
Same comment from other one
"I think it'd be better if this functionality was abstracted out to another function, as it's not super clear from this line of code exactly what this is doing. Maybe a function like a.Config.RemovePlugin(pluginName string) or something? Same for add/update plugins. And then eventually this would need to be migrated from using pluginNames to using the plugins uuids."